vm/math.cpp

```diff
@@ -217,6 +219,40 @@ void factor_vm::primitive_format_float() {
   ctx->replace(tag<byte_array>(array));
 }
 
+/* Allocates memory */
+void factor_vm::primitive_format_float_C() {
+  char* format = alien_offset(ctx->pop());
+  fixnum precision = untag_fixnum(ctx->pop());
+  fixnum width = untag_fixnum(ctx->pop());
+  char* fill = alien_offset(ctx->pop());
+  double value = untag_float_check(ctx->peek());
+  std::ostringstream locale_C_stream;
+  locale_C_stream.imbue(std::locale("C"));
+  switch (format[0]) {
+    case 'F': locale_C_stream << std::uppercase;
+    case 'f': locale_C_stream << std::fixed; break;
+    case 'E': locale_C_stream << std::uppercase;
+    case 'e': locale_C_stream << std::scientific; break;
+    case 'G': locale_C_stream << std::uppercase;
+  }
```

Comment:
I think `if (isupper(format[0])) { locale_C_stream << std::uppercase; }` is clearer here than using the break-fallthrough trick.